src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -248,51 +248,54 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         // Subscribe to some changes in the view model
         // These changes should force us to update our own set of "Current<Setting>" members,
         // and propagate those changes to the UI
-        _State.Profile().PropertyChanged([&](auto&&, const PropertyChangedEventArgs& args) {
-            if (args.PropertyName() == L"CursorShape")
+        _ViewModelChangedRevoker = _State.Profile().PropertyChanged(winrt::auto_revoke, [weakThis{ get_weak() }](auto&&, const PropertyChangedEventArgs& args) {
+            if (auto page{ weakThis.get() })
             {
```

Comment:
no need for a weak capture here. the revoker ensures that either events are delivered to a living instance, or the registration is revoked